show/vlan.py

```diff
@@ -110,32 +110,29 @@ def config(db):
     keys = list(data.keys())
     member_data = db.cfgdb.get_table('VLAN_MEMBER')
 
+    def member_name(member):
+        alias = member
+        if clicommon.get_interface_naming_mode() == "alias" and member:
+            alias = clicommon.InterfaceAliasConverter(db).name_to_alias(member)
+        return alias
+    
+    def member_tagging(vlan, member):
+        if not member:
+            return ''
+        mode = db.cfgdb.get_entry('VLAN_MEMBER', (vlan, member))['tagging_mode']
+        return '?' if mode is None else mode
+        
     def tablelize(keys, data):
         table = []
 
         for k in natsorted(keys):
-            members = set(data[k].get('members', []))
-            for (vlan, interface_name) in member_data:
-                if vlan == k:
-                    members.add(interface_name)
-
-            for m in natsorted(list(members)):
-                r = []
-                r.append(k)
-                r.append(data[k]['vlanid'])
-                if clicommon.get_interface_naming_mode() == "alias":
-                    alias = clicommon.InterfaceAliasConverter(db).name_to_alias(m)
-                    r.append(alias)
-                else:
-                    r.append(m)
-
-                entry = db.cfgdb.get_entry('VLAN_MEMBER', (k, m))
-                mode = entry.get('tagging_mode')
-                if mode is None:
-                    r.append('?')
-                else:
-                    r.append(mode)
-
+            members = [(vlan, member) for vlan, member in member_data if vlan == k]
```

Comment:
I think the previous code also checked for 'members' field. Looks like you are skipping it?

Reply:
Yeah, this also appears to remove any interface name <--> alias translation. Can you explain why you removed the code that you did?

Reply:
The problem with the previous code was that if there are no members the code 
           ` for m in natsorted(list(members)):`
was skipped.
This was main reason that VLAN with no members wasn't appeared in `show vlan config`
The tuple of (Vlan, member) for each member is built in 
`members = [(vlan, member) for vlan, member in member_data if vlan == k]`
or if no members 
members=[(vlan, '')]
Now for every tuple we build record:
`['Name' = vlan, 'VID' = data[vlan]['vlanid'], 'Member' = member_name(member), 'Mode' = member_tagging(vlan,member)]`
Function `member_name` will translate interface name to alias (if required) and function member_tagging will return tagging (if required = if member exists).
In addition, unit tests for vlan were passed.

Reply:
I think I got your point. You talked about config_db.json ....
You are right - with config_db.json its not working - I will fix it.
I noticed that show vlan brief is not representing members for DB configured in config_db.json: 
` show vlan brief `
```
+-----------+--------------+---------+----------------+-----------------------+-------------+
|   VLAN ID | IP Address   | Ports   | Port Tagging   | DHCP Helper Address   | Proxy ARP   |
+===========+==============+=========+================+=======================+=============+
|       100 |              |         |                |                       | disabled    |
+-----------+--------------+---------+----------------+-----------------------+-------------+
 
```
when in the previous version for
`show vlan config`
```
Name       VID  Member     Mode
-------  -----  ---------  ------
Vlan100    100  Ethernet0  ?

```
I see its as a BUG.
What do you think?

Reply:
> I think I got your point. You talked about config_db.json ....

Who talked about config_db.json?

Also, you did not address the removal of interface name <--> alias translation.